### ISSUE TITLE
Go 2: Make life easier

### ISSUE THREAD

**Issue by pjebs** [golang/go#28760]
Suppose I have a variable:

```
Example 1:

var x *int
x = &5 // This fails.

Example 2:

var t *time.Time
t = &time.Now() // this fails
```

The recommended way is:

```

var x *int

var dummy int = 5 
x = &dummy // This works

Or

var t *time.Time

var dummy time.Time = time.Now()
t = &dummy // This works
```

**This is annoying. Having to create dummy variables just to do this.**

There is also an "apparent" inconsistency (despite following similar semantics to above):

```
type A struct {}

var x *A

x = &A{} // This is allowed
```

**Comment by pjebs** [comment]
I see this dummy variable style absolutely everywhere in goland. It looks disgusting. It's still less revolting that the temporary slice (which is what I use everywhere)

**Comment by randall77** [comment]
Dup of #9097 (and more directly of #19966, which was closed in favor of #9097). Closing as a dup.

**Comment by pjebs** [comment]
I've created https://github.com/rocketlaunchr/igo to demo how it can be used.
